packages/i18n/src/messages/localizedConstants.tsx

```diff
@@ -1597,7 +1597,7 @@ export const poolCandidatePriorities = defineMessages({
 });
 
 export const getPoolCandidatePriorities = (
-  priorityWeight: string | number,
+  priorityWeight: string,
```

Comment:
This will never have a number input, we've been doing this for a while as the format. And when we do input a number like priority weight it errors. 
I constricted it just for this one function and it revealed another instance of this issue's bug.

![image](https://github.com/GCTC-NTGC/gc-digital-talent/assets/40485260/2221d8cc-a7fe-4f66-8284-ecbc4633a4bb)